File: csharp/raindrops/Raindrops.cs
namespace Exercism_raindrops
{
    public class Raindrops
    {
        public static string Convert(int number)
        {
            string convertResult = string.Empty;

            if (number % 3 == 0)
            {
                convertResult += "Pling";
            }

            if (number % 5 == 0)
            {
                convertResult += "Plang";
            }

            if (number % 7 == 0)
            {
                convertResult += "Plong";
            }

            return convertResult == string.Empty ? number.ToString() : convertResult;
        }
    }
}

File: javascript/raindrops/raindrops.js
class Raindrops {
    constructor() {}

    convert(number) {
        let output = [];

        if (number % 3 === 0) {
            output.push("Pling");
        }

        if (number % 5 === 0) {
            output.push("Plang");
        }

        if (number % 7 === 0) {
            output.push("Plong")
        }

        if (output.length === 0) {
            output.push(number + '');
        }

        return output.join('');
    }
};

module.exports = Raindrops;
